internal/server/config.go

```diff
@@ -20,87 +20,87 @@ import (
 )
 
 type Provider struct {
-	Name         string `mapstructure:"name" validate:"required"`
-	URL          string `mapstructure:"url" validate:"required"`
-	ClientID     string `mapstructure:"clientID" validate:"required"`
-	ClientSecret string `mapstructure:"clientSecret" validate:"required"`
+	Name         string `validate:"required"`
+	URL          string `validate:"required"`
+	ClientID     string `validate:"required"`
+	ClientSecret string `validate:"required"`
 }
 
 type Grant struct {
-	User     string `mapstructure:"user" validate:"excluded_with=Group,excluded_with=Machine"`
-	Group    string `mapstructure:"group" validate:"excluded_with=User,excluded_with=Machine"`
-	Machine  string `mapstructure:"machine" validate:"excluded_with=User,excluded_with=Group"` // deprecated
-	Resource string `mapstructure:"resource" validate:"required"`
-	Role     string `mapstructure:"role"`
+	User     string `validate:"excluded_with=Group,excluded_with=Machine"`
+	Group    string `validate:"excluded_with=User,excluded_with=Machine"`
+	Machine  string `validate:"excluded_with=User,excluded_with=Group"` // deprecated
+	Resource string `validate:"required"`
+	Role     string
 }
 
 type User struct {
-	Name      string `mapstructure:"name" validate:"excluded_with=Email"`
-	AccessKey string `mapstructure:"accessKey"`
-	Password  string `mapstructure:"password"`
+	Name      string `validate:"excluded_with=Email"`
+	AccessKey string
+	Password  string
 
-	Email string `mapstructure:"email" validate:"excluded_with=Name"` // deprecated
+	Email string `validate:"excluded_with=Name"` // deprecated
 }
 
 type Config struct {
-	Providers []Provider `mapstructure:"providers" validate:"dive"`
-	Grants    []Grant    `mapstructure:"grants" validate:"dive"`
-	Users     []User     `mapstructure:"users" validate:"dive"`
+	Providers []Provider `validate:"dive"`
+	Grants    []Grant    `validate:"dive"`
+	Users     []User     `validate:"dive"`
 }
 
 type KeyProvider struct {
-	Kind   string      `mapstructure:"kind" validate:"required"`
+	Kind   string      `validate:"required"`
```

Comment:
The key provider and secret provider loading is covered by:
* `server.TestKeyProvider_PrepareForDecode_IntegrationWithDecode_FullConfig`
* `server.TestSecretProvider_PrepareForDecode_IntegrationWithDecode_FullConfig`